### PR TITLE
feat(DatePicker, DatePickerInput): refactor class component to functional component

### LINKED ISSUE #10009: [DatePicker]: upgrade to functional component

### LINKED ISSUE #10010: [DatePickerInput]: upgrade to functional component

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #10010
Closes #10009
ref #9712 

#### Changelog

- refactor to class component 
- no tests for this component

#### Testing / Reviewing

- Test carbon-react-next date picker to make sure styles and logic is working as expected

### CODE REVIEW

File: packages/react/src/components/DatePickerInput/next/DatePickerInput.js [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        onClick(evt);
+      }
+    },
+    placeholder,
+    type,
+    pattern,
+  };
+
+  const wrapperClasses = classNames(
+    `${prefix}--date-picker-input__wrapper NEXT-WRAPPER`,
```
Comment (jnm2377) [inline packages/react/src/components/DatePickerInput/next/DatePickerInput.js:55]: note for reviewers: the class `NEXT-WRAPPER` is just for testing purposes. To make sure it's using the next component. will remove before merging.

**Review by tay1orjones — COMMENTED** [review]
@jnm2377 did the existing tests blow up pretty hard against the `/next` version?

**Comment by jnm2377** [comment]
@tay1orjones I didn't check the test tbh. But the `DatePickerInput` folder actually didn't have any tests at all.

**Comment by tay1orjones** [comment]
@jnm2377 Ah yeah that makes sense, my bad. This'll get validated moreso through the DatePicker refactor https://github.com/carbon-design-system/carbon/issues/10009

**Comment by abbeyhrt** [comment]
@jnm2377 I'm not sure how this would be from your PR but I'm not able to get the v11 storybook running locally for the PR and the preview is failing because of a time out, is it working for you locally?

**Comment by jnm2377** [comment]
@abbeyhrt @tay1orjones seems like the build didn't like the icon imports. Just pushed a fix. Should be ready for review. :)

**Comment by abbeyhrt** [comment]
@jnm2377 I'm not sure the relationship between `DatePickerInput` and and the calendar that drops down but I'm not able to get the calendar to show up in the storybook 😕

**Comment by jnm2377** [comment]
@abbeyhrt hmmm... I wonder if I should make the `DatePicker` changes in this PR too... the `DatePicker` component does use `DatePickerInput` and that's maybe why it's not working right. I haven't made those changes here. This refactor is essentially only for the simple input, not the calendar variation.

**Comment by jnm2377** [comment]
closing in favor of https://github.com/carbon-design-system/carbon/pull/10156

**Pull request closed without merging**